### DISCUSSION TITLE
DataDraw is interesting

### DISCUSSION THREAD

**Discussion by dibyendumajumdar** [google/rune#23] [General]
Hi

I found the use of DataDraw to generate C code from https://github.com/google/rune/blob/main/database/Rune.dd interesting.

I was wondering if I could use this in my own language implementation, but I wasn't sure if nested scopes can be represented in DataDraw.

Regards

**Comment by waywardgeek** [comment]
Nested scopes are easy to represent:


    class Scope

    relationship Scope Scope doubly_linked cascade

This creates a doubly-linked list of scopes on scopes.  Feel free to use an array relationship if that feels more comfortable.

**Comment by dibyendumajumdar** [comment]
Okay I see - thank you
